src/dmd/parse.d

```diff
@@ -5094,6 +5094,10 @@ final class Parser(AST) : Lexer
         if (token.value == TOK.goesTo)
         {
             check(TOK.goesTo);
+            if (token.value == TOK.leftCurly)
+            {
+                deprecation("=> { ... } is a lambda that returns a delegate. Use (args) { ... } for a multi-statement function literal or use => () { } if you intended to return a delegate.");
```

Comment:
Use `deprecationSupplemental` for the suggestion.
Also, changelog.
Also, test.

But I thought this was warned against already, maybe in semantic ?

Reply:
There's no warning now, this has tripped up dozens of people.

Reply:
this has nothing to do in a parser. too much expectactions on the semantics.

Reply:
@adamdruppe either way it still requires at least a test and changelog.